shopinvader_product/models/product_product.py

```diff
@@ -64,34 +64,51 @@ def _get_shopinvader_product_variants(self, product_ids):
             [("product_tmpl_id", "in", product_ids)], order="product_tmpl_id"
         )
 
+    @api.model
+    def _get_main_product_read_fields(self):
+        product_model = self.env["product.product"]
+        order_by = [x.strip() for x in product_model._order.split(",")]
+        return ["product_tmpl_id"] + [f.split(" ")[0] for f in order_by]
+
+    @api.model
+    def _get_main_product_sorted_variants(self, variants):
+        # NOTE: if the order is changed by adding `asc/desc` this can be broken
+        # but it's very unlikely that the default order for product.product
+        # will be changed.
+        order_by = [x.strip() for x in self.env["product.product"]._order.split(",")]
```

Comment:
`order_by` used to be the same as the one used to compute `fields_to_read`. I'm a bit lost, but shouldn't we keep this consistency ? @simahawk ?

Reply:
You would like to extract the compute of `order_by` from `_get_main_product_read_fields` into a new hook to re-use it here into `_get_main_product_sorted_variants` right?

Reply:
If necessary, yes.